### DISCUSSION TITLE
forge本地搭建出现显卡CUDA Torch不兼容问题已解决

### DISCUSSION THREAD

**Discussion by boboHenry** [lllyasviel/stable-diffusion-webui-forge#2469] [Q&A] [answered]
![QQ截图20241216232807](https://github.com/user-attachments/assets/8795f029-cbf0-43ce-95b3-021bb49c24b7)
哪位大佬告诉我这个怎么解决啊，我是新手正在学习中

**Comment by boboHenry** [comment]
我的显卡信息
![QQ截图20241216232957](https://github.com/user-attachments/assets/da2cb14a-d51f-41d9-84ec-b7521ca81840)

**Comment by zeittresor** [comment]
你好。我不确定，但也许您应该更新 CUDA 驱动程序或尝试不同的版本。我的 Nvidia Tesla K80 附加显卡也有类似的问题。

**Reply by boboHenry** [reply]
第一步（安装你电脑的显卡驱动）
![333](https://github.com/user-attachments/assets/e0f464ef-2237-4e0b-b713-90b0545dfe7a)

这个问题我解决了，首先如果你是Windows系统打开cmd，查看你的显卡版本信息（命令nvidia-smi此命令在你安装了你的显卡对应驱动后才可以使用），我的显卡版本是 Driver Version: 475.14。去nvidia官网https://www.nvidia.cn/drivers/lookup/下载对应的驱动版本。
![01](https://github.com/user-attachments/assets/ab9ed8c0-316d-46a2-8ad9-9f5c5800f454)
![02](https://github.com/user-attachments/assets/e4266e0e-6815-4e33-bef4-1f374e9e542c)

第二步（下载安装你电脑显卡对应的CUDA）
CUDA下载地址https://developer.nvidia.com/cuda-toolkit-archive
![image](https://github.com/user-attachments/assets/7e29500d-2ea6-4592-8585-88c4cf5b9391)
安装好CUDA后使用nvcc -V查看安装是否成功以及对应的CUDA版本，我电脑CUDA安装的版本是11.4
![image](https://github.com/user-attachments/assets/41529b9e-2c92-492c-b0a4-be82d064961b)

第三步（安装PyTorch 地址：https://pytorch.org/get-started/previous-versions/）
不要安装最新版PyTorch会有兼容问题，要安装你电脑安装的CUDA版本对应的PyTorch，我这里安装的是# CUDA 11.6版本
![image](https://github.com/user-attachments/assets/df08eef9-cffa-4e08-9321-ffbdc96bab59)

以上三步就可以解决forge运行报显卡版本错误以及CUDA错误，以及CUDA对应的torch不兼容问题

**Reply by boboHenry** [reply, marked as answer]
第一步（安装你电脑的显卡驱动）
![333](https://github.com/user-attachments/assets/e0f464ef-2237-4e0b-b713-90b0545dfe7a)

这个问题我解决了，首先如果你是Windows系统打开cmd，查看你的显卡版本信息（命令nvidia-smi此命令在你安装了你的显卡对应驱动后才可以使用），我的显卡版本是 Driver Version: 475.14。去nvidia官网https://www.nvidia.cn/drivers/lookup/下载对应的驱动版本。
![01](https://github.com/user-attachments/assets/ab9ed8c0-316d-46a2-8ad9-9f5c5800f454)
![02](https://github.com/user-attachments/assets/e4266e0e-6815-4e33-bef4-1f374e9e542c)

第二步（下载安装你电脑显卡对应的CUDA）
CUDA下载地址https://developer.nvidia.com/cuda-toolkit-archive
![image](https://github.com/user-attachments/assets/7e29500d-2ea6-4592-8585-88c4cf5b9391)
安装好CUDA后使用nvcc -V查看安装是否成功以及对应的CUDA版本，我电脑CUDA安装的版本是11.4
![image](https://github.com/user-attachments/assets/41529b9e-2c92-492c-b0a4-be82d064961b)

第三步（安装PyTorch 地址：https://pytorch.org/get-started/previous-versions/）
不要安装最新版PyTorch会有兼容问题，要安装你电脑安装的CUDA版本对应的PyTorch，我这里安装的是# CUDA 11.6版本
![image](https://github.com/user-attachments/assets/df08eef9-cffa-4e08-9321-ffbdc96bab59)

以上三步就可以解决forge运行报显卡版本错误以及CUDA错误，以及CUDA对应的torch不兼容问题

**Reply by boboHenry** [reply]
一定要把你电脑以前安装的CUDA卸载掉

**Comment by boboHenry** [comment]
今天我测试了一下还是我的台式机显卡问题，我台式机显卡不支持安装stable-diffusion-webui-forge，我换了笔记本RTX1050 4G显卡就能成功安装
